source/common/upstream/cds_api_impl.cc

```diff
@@ -53,9 +54,13 @@ void CdsApiImpl::onConfigUpdate(const ResourceVector& resources, const std::stri
   ClusterManager::ClusterInfoMap clusters_to_remove = cm_.clusters();
   for (auto& cluster : resources) {
     const std::string cluster_name = cluster.name();
-    clusters_to_remove.erase(cluster_name);
-    if (cm_.addOrUpdateCluster(cluster, version_info)) {
-      ENVOY_LOG(debug, "cds: add/update cluster '{}'", cluster_name);
+    try {
+      clusters_to_remove.erase(cluster_name);
+      if (cm_.addOrUpdateCluster(cluster, version_info)) {
+        ENVOY_LOG(debug, "cds: add/update cluster '{}'", cluster_name);
+      }
+    } catch (const EnvoyException& e) {
+      exceptions.push_back(e);
```

Comment:
drive-by c++ comment: 
assigning a reference to a value (as done here: `exceptions.push_back(e);`) is generally undesirable. if the exception is of a sub-type of EnvoyException, the object will be 'cut' and only the base object attributes will be copied.
In this case you are only using the messages (`exception.what()`), I would suggest using a list of strings (`std::list<std::string> exceptions;`), instead of EnvoyExceptions, and do `exceptions.push_back(e.what())`

Reply:
Good point. We will rework.